Extension/src/LanguageServer/extension.ts

```diff
@@ -250,35 +250,41 @@ async function installVsix(vsixLocation: string, updateChannel: string): Promise
                 execSync(uninstallCommand);
             }
             execSync(installCommand);
+            clearInterval(insiderUpdateTimer);
             return Promise.resolve();
         } catch (error) {
-            return Promise.reject(new Error('Install command failed'));
+            return Promise.reject(new Error('Failed to install vsix'));
         }
     });
 }
 
 async function checkAndApplyUpdate(updateChannel: string): Promise<void> {
-    return getTargetBuildURL(updateChannel).then((downloadUrl) => {
-        if (!downloadUrl) {
-            return Promise.resolve();
-        }
-
-        // Create a temporary file, download the vsix to it, then install the vsix
-        tmp.file({postfix: '.vsix'}, async (err, vsixPath, fd, cleanupCallback) => {
-            if (err) {
-                cleanupCallback();
-                return Promise.reject(new Error('Failed to create vsix file'));
+    const p: Promise<void> = new Promise<void>((resolve, reject) => {
+        getTargetBuildUrl(updateChannel).then((downloadUrl) => {
+            if (!downloadUrl) {
+                return resolve();
             }
-            await util.downloadFileToDestination(downloadUrl, vsixPath).catch(() => {
-                cleanupCallback();
-                return Promise.reject(new Error('Failed to download vsix'));
+
+            // Create a temporary file, download the vsix to it, then install the vsix
+            tmp.file({postfix: '.vsix'}, (err, vsixPath, fd, cleanupCallback) => {
+                if (err) {
+                    return reject(new Error('Failed to create vsix file'));
+                }
+    
+                const downloadVsix: any = util.downloadFileToDestination.bind(undefined, downloadUrl, vsixPath);
+                const boundInstallVsix: any = installVsix.bind(undefined, vsixPath, updateChannel);
+                const logSuccess: any = () => { telemetry.logLanguageServerEvent('installVsix', { 'Success': 'true' } ); };
```

Comment:
Please don't use `bind`.  I thought we talked about not using `.then(func)` syntax in our code (which seems to be the underlying reason for this).  Please use `var => func(var)` syntax for the continuation clauses.  It makes the code easier to read and maintain.